src/framework/components/anim/system.js

```diff
@@ -35,16 +32,26 @@ class AnimComponentSystem extends ComponentSystem {
     }
 
     initializeComponentData(component, data, properties) {
-        properties = ['activate', 'enabled', 'speed', 'playing'];
-        super.initializeComponentData(component, data, properties);
-        if (data.stateGraphAsset) {
-            component.stateGraphAsset = data.stateGraphAsset;
+        properties = ['activate', 'speed', 'playing'];
+        super.initializeComponentData(component, data, _schema);
+        Object.keys(data).forEach((key) => {
+            if (['animationAssets', 'stateGraph', 'layers'].includes(key)) return;
```

Comment:
Does the array here get allocated each loop? 

`['animationAssets', 'stateGraph', 'layers']`

Reply:
I tend to keep short lists in line as it makes them readable to me but I can lift it out if you have a concern.

Reply:
I think it be good to pull it out if it's allocated each iteration

Reply:
Done 👍